go/test/endtoend/vtgate/queries/aggregation/aggregation_test.go

```diff
@@ -69,6 +70,17 @@ func start(t *testing.T) (utils.MySQLCompare, func()) {
 	}
 }
 
+func TestAggrWithLimit(t *testing.T) {
+	mcmp, closer := start(t)
+	defer closer()
+
+	for i := range 1000 {
+		r := rand.IntN(10)
+		mcmp.Exec(fmt.Sprintf("insert into aggr_test(id, val1, val2) values(%d, 'a', %d)", i, r))
+	}
+	mcmp.Exec("select val2, count(*) from aggr_test group by val2 order by count(*), val2 limit 10")
+}
```

Comment:
This test always pass with and without the fix proposed by this PR. Same thing happens on the original PR (https://github.com/vitessio/vitess/pull/16263)